ipn/ipnlocal/local.go

```diff
@@ -1788,7 +1788,9 @@ func (b *LocalBackend) setPrefsLockedOnEntry(caller string, newp *ipn.Prefs) {
 	b.send(ipn.Notify{Prefs: newp})
 }
 
-func (b *LocalBackend) getPeerAPIPortForTSMPPing(ip netaddr.IP) (port uint16, ok bool) {
+// GetPeerAPIPort returns the port number for the peerapi server
+// running on the provided IP.
+func (b *LocalBackend) GetPeerAPIPort(ip netaddr.IP) (port uint16, ok bool) {
```

Comment:
So we bind to port 0 and let the kernel pick a port, but then pick off incoming packets to the port in netstack?

The main thing I'm trying to think through is if there could be a case where we close the kernel socket, potentially allowing some other process to bind to that same ephemeral port, but we neglect to cleanup the netstack listener and end up intercepting traffic which doesn't belong to us.

Reply:
that's why on every incoming SYN to netstack, we double-check that the PeerAPIPort is accurate.

Reply:
> So we bind to port 0 and let the kernel pick a port, but then pick off incoming packets to the port in netstack?

Yes.